sextant/src/main.rs

```diff
@@ -66,9 +79,455 @@ enum Opt {
     },
 }
 
+// TODO: all from here: move to its own file
+
+const DIGEST_SIZE: usize = 32;
+const BLOCK_SIZE: u64 = 4096;
+
 fn main() -> Result<()> {
     env_logger::init();
     let opt = Opt::from_args();
     info!("{:#?}", opt);
-    todo!();
+    match opt {
+        Opt::Pack { dir, out } => pack_cmd(&dir, &out),
+        _ => {
+            unimplemented!();
+        }
+    }
+}
+
+#[derive(Copy, Clone, PartialEq)]
+enum FsType {
+    SQUASHFS,
+    EXT4,
+}
+
+fn pack_cmd(_dir: &Path, _out: &Path) -> Result<()> {
+    let current_dir = std::env::current_dir()?;
+    let example_dir = current_dir.join("examples");
+    let container_src_dir = example_dir.join("container");
+    let key_dir = example_dir.join("keys");
+    let registry_dir = current_dir.join("target").join("north").join("registry");
+
+    pack_containers(
+        &registry_dir,
+        &container_src_dir,
+        &key_dir,
+        "north",
+        FsType::SQUASHFS,
+        1000,
+        1000,
+    )
+}
+
+fn pack_containers(
+    registry_dir: &Path,
+    container_src_dir: &Path,
+    key_dir: &Path,
+    signing_key_name: &str,
+    fs_type: FsType,
+    uid: u32,
+    gid: u32,
+) -> Result<()> {
+    log::debug!("");
+    log::debug!("pack_containers called with");
+    log::debug!("registry_dir={}", registry_dir.display());
+    log::debug!("container_src_dir={}", container_src_dir.display());
+    log::debug!("key_dir={}", key_dir.display());
+    log::debug!("signing_key_name={}", signing_key_name);
+
+    let sign_key_path = key_dir.join(signing_key_name).with_extension("key");
+    log::debug!("open file {}", sign_key_path.display());
+    let sign_key = File::open(&sign_key_path)?;
+    log::debug!("sign_key.len()={}", sign_key.metadata().unwrap().len());
```

Comment:
no `unwrap()` should remain in the code. If you use this in a function that returns a result, you should be able to use the `?` operator